src/staging/src/components/park/parkFacility.js

```diff
@@ -11,61 +12,111 @@ import StaticIcon from "./staticIcon"
 import "../../styles/cmsSnippets/parkInfoPage.scss"
 
 export default function ParkFacility({ data }) {
-  const facilityData = data
-  const [expanded, setExpanded] = useState(
-    Array(facilityData.length).fill(false)
+  const [facilityData, setFacilityData] = useState(
+    JSON.parse(JSON.stringify(data)) // deep copy
   )
+  const [expanded, setExpanded] = useState(Array(data.length).fill(false))
+  const [numFacilities, setNumFacilities] = useState(0)
 
-  if (facilityData.length === 0) return null
+  const toggleExpand = useCallback(
+    index => {
+      expanded[index] = !expanded[index]
+      setExpanded([...expanded])
+    },
+    [expanded]
+  )
+  //const [isInitial, setIsInitial] = useState(true)
+
+  const checkHash = useCallback(() => {
+    // Check hash in url
+    // if we find a matching facilityCode, open that facility accordion
+    // TODO setup useEffect etc.
+    let h = ""
+
+    if (typeof window !== "undefined") {
+      h = window.location.hash
+      if (h !== undefined) {
+        for (const idx in facilityData) {
+          const facility = facilityData[idx]
+          if (h === "#" + facility.facilityType.facilityCode) {
+            if (!expanded[facility.uiId]) {
+              toggleExpand(facility.uiId)
+            }
+          }
+        }
+      }
+    }
+  }, [expanded, facilityData, toggleExpand])
 
-  const toggleExpand = index => {
-    expanded[index] = !expanded[index]
-    setExpanded([...expanded])
-  }
+  useEffect(() => {
+    // Each element in the array needs a unique id for the accordions
+    // to be togglable through code
+    let i = 0
+    for (const idx in facilityData) {
```

Comment:
I don't think it causes any problems here but for...in is not ideal for arrays, just FYI: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#array_iteration_and_for...in

probably a `forEach` case or standard for loop

Reply:
Done.